sympy/solvers/decompogen.py

```diff
@@ -37,17 +37,25 @@ def decompogen(f, symbol):
     if isinstance(f, (Function, Pow)):
         if f.args[0] == symbol:
             return [f]
-        result += [f.subs(f.args[0], symbol)] + decompogen(f.args[0], symbol)
+        try:
+            result += [f.subs(f.args[0], symbol)] + decompogen(f.args[0], symbol)
+        except TypeError:
+            result = [f]
         return result
 
     # ===== Convert to Polynomial ===== #
-    fp = Poly(f)
+    try:
+        fp = Poly(f)
+    except:
```

Comment:
`symbol` is an argument of decompogen. That should be used as the generator (second argument of Poly).